src/dotty/tools/dotc/transform/ClassTags.scala

```diff
@@ -0,0 +1,57 @@
+package dotty.tools.dotc
+package transform
+
+import core._
+import TreeTransforms._
+import Contexts.Context
+import Flags._
+import SymUtils._
+import Symbols._
+import SymDenotations._
+import Types._
+import Decorators._
+import DenotTransformers._
+import StdNames._
+import NameOps._
+import ast.Trees._
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.core.Constants.Constant
+import util.Positions._
+import Names._
+import collection.mutable
+
+/** This phase replaces calls to DottyPredef.classTag by code that synthesizes appropriate ClassTag
+  */
+class ClassTags extends MiniPhaseTransform with IdentityDenotTransformer { thisTransform =>
+  import ast.tpd._
+
+  object names {
+    val classTag = "classTag".toTermName
```

Comment:
It is specific to this phase, no other phase needs those names.

We already have a lot of names in `nme` which are not used, if instead of having them in `nme` they were defined in respective phases, they would have been removed when stopped being used by phase.

Though having all names in same place could also be a valuable argument. If you thing so, should I also move names from `LazyVals` to `nme`?